### PR TITLE
Add backup configurations

### PR DESCRIPTION
**Pull request by WadeBarnes**
- Copied over and updated from https://github.com/bcgov/jag-shuber-api

Signed-off-by: Wade Barnes <wade.barnes@shaw.ca>

### CODE REVIEW

**Comment by WadeBarnes** [comment]
Build configuration updated and tested.  Needed to upgrade to [BCDevOps/backup-container - 2.2.1](https://github.com/BCDevOps/backup-container/releases/2.2.1) for Postgres 12 support.

Deployment configurations updated and tested in `dev` and `test`.  Leaving `prod` environment alone for now.

File: openshift/templates/backup/backup-deploy.prod.param [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ENVIRONMENT_NAME=jag-shuber-prod
+# BACKUP_DIR=/backups/
+# DAILY_BACKUPS=
+# WEEKLY_BACKUPS=
+# MONTHLY_BACKUPS=
+# DATABASE_SERVER_TIMEOUT=120
+# CONFIG_FILE_NAME=backup.conf
+# CONFIG_MAP_NAME=backup-conf
+# CONFIG_MOUNT_PATH=/
+BACKUP_VOLUME_NAME=bk-jag-shuber-prod-kcupbl9x9rx0
```
Comment (devinleighsmith) [inline openshift/templates/backup/backup-deploy.prod.param:28]: Is this reusing a pre-existing volume? I just noticed that bk-jag-shuber-prod-kcupbl9x9rx0 was created 4 months ago

File: openshift/templates/backup/backup-deploy.json [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+                  {
+                    "name": "POSTGRESQL_PASSWORD",
+                    "valueFrom": {
+                      "secretKeyRef": {
+                        "name": "${POSTGRESQL_SERVICE_NAME}${SUFFIX}",
+                        "key": "${POSTGRESQL_PASSWORD_KEY_NAME}"
+                      }
+                    }
+                  },
+
```
Comment (devinleighsmith) [inline openshift/templates/backup/backup-deploy.json]: odd whitespace issue.

File: openshift/templates/backup/backup-deploy.prod.param (hunk 1 above)
Comment (WadeBarnes) [reply openshift/templates/backup/backup-deploy.prod.param:28]: Yes, it's reusing (sharing) the existing volume.

**Review by devinleighsmith — APPROVED** [review]

File: openshift/templates/backup/backup-deploy.json (hunk 2 above)
Comment (WadeBarnes) [reply openshift/templates/backup/backup-deploy.json]: Addressed

**Pull request merged**